inst/pages/multimedia.qmd

```diff
@@ -0,0 +1,590 @@
+analysis, which typically focuses on a single mediator or a small set of 
+mediators, we performed multimodal mediation analysis with R/Bioconductor 
+package \[multimedia\] [@Jiang2025] here, which can handle many potential 
+mediators across multiple data modalities. For example, species-level taxonomic 
+abundances, functional pathways, and metabolomic profiles can all serve as 
+potential mediators. This method allows us to estimate and rank each mediator’s 
+contribution while accounting for correlations between mediators in a 
+high-dimensional setting.
+
+In this example, time point (baseline vs. post-treatment) is used as the 
+treatment, and microbiome-based dysbiosis scores are used as the outcome. We 
+will first focus on the relative abundances of gut microbial species as 
+mediators, followed by functional pathways.
+
+Generally, we will proceed through the following key steps:
+
+1.  Prepare the microbiome taxonomic data from the iHMP IBDMDB project.
+
+2.  Calculate dysbiosis scores as the outcome.
+
+3.  Define the mediation analysis data structure.
+
+4.  Fit the multimodal mediation model using the R package multimedia.
+
+5.  Interpret both the overall indirect effects and the mediator-specific 
+indirect effects.
+
+6.  Visualize the results using forest plots, histograms, and rankings to 
+prioritize findings.
+
+7.  Repeat the process for the iHMP microbial pathways.
+
+### Performing multimodal mediation analysis for iHMP species relative abundance {#sec-relative-abundance}
+
+We begin by loading the relative abundance data and calculating dysbiosis scores 
+for each sample based on Bray–Curtis dissimilarity from a healthy reference set. 
+This represents deviation from a healthy microbiome, where higher dysbiosis 
+scores indicate greater microbial imbalance. We then subset to IBD subjects with 
+samples collected at both baseline (visit 1) and post-treatment (visit 21).
+
+```{r}
+##################
+# Load libraries #
+##################
+
+library(curatedMetagenomicData)
+library(SummarizedExperiment)
+library(dplyr)
+library(vegan)
+library(tidyverse)
+library(multimedia)
+library(stringr)
+library(ggplot2)
+
+
+##################
+# Load iHMP data #
+##################
+
+# Load data
+se_relative <- curatedMetagenomicData(
+  "HMP_2019_ibdmdb.relative_abundance", 
+  dryrun = FALSE
+  )[[1]]
+
+# Normalize to proportions (from %)
+assay(se_relative) <- assay(se_relative) / 100
+
+# Change the rownames names for variables of interest
+rownames(se_relative) <- sub('.*s__', '', rownames(se_relative))
+
+
+########################
+# Reference nonIBD set #
+########################
+
+# Reference set: healthy
+ref_set <- colData(se_relative)$disease == "healthy"
+
+
+########################################
+# Calculate Bray-Curtis dissimilarity #
+########################################
+
+# Bray-Curtis dissimilarity
+dist <- as.matrix(vegdist(t(assay(se_relative)), method = "bray", na.rm = TRUE))
```

Comment:
I would rename "dist" to "diss" because Bray-Curtis is not a distance.

We also use ready-made mia function available to add dissimilarities on the data object. It calls vegan but is more directly compatible with the rest of the downstream tools in the Bioc ecosystem.

See: 
- [mia::getDissimilarity](https://microbiome.github.io/mia/reference/getDissimilarity.html) and 
- [mia::addDissimilarity](https://microbiome.github.io/mia/reference/addDissimilarity.html)

Reply:
I think you could replace the above with:

`res <- getDissimilarity(tse, method = "bray", na.rm=TRUE, assay.type = "relabundance")`

Then to get it as a matrix you  can do:
`as.matrix(res) `